### DISCUSSION TITLE
Issue with submodules

### DISCUSSION THREAD

**Discussion by Mearman** [Vinzent03/obsidian-git#438] [Q&A]
Hey there.

your plugin is one I install in EVERY notebook because it is superb.
however I've got a case where I need one notebook as a submodule of the other. 
for some reason I'm getting the following error . I'm wondering if something is wrong in the syntax of my `.gitmodules` file? It works fine with regular git clients?

I was also wondering if there is an options hidden away somewhere to still allow push of the main repo even if a submodule push fails?

```
Uncaught (in promise) Error: fatal: No url found for submodule path 'Projects/Personal/Steam Redeem/Steam Redeem Gist' in .gitmodules

    at Object.action (plugin:obsidian-git:23985:25)
    at PluginStore.exec (plugin:obsidian-git:24010:25)
    at eval (plugin:obsidian-git:21571:43)
    at new Promise (<anonymous>)
    at GitExecutorChain.handleTaskData (plugin:obsidian-git:21569:16)
    at GitExecutorChain.eval (plugin:obsidian-git:21553:44)
    at Generator.next (<anonymous>)
    at fulfilled (plugin:obsidian-git:20402:24)
action @ plugin:obsidian-git:23985
exec @ plugin:obsidian-git:24010
eval @ plugin:obsidian-git:21571
handleTaskData @ plugin:obsidian-git:21569
eval @ plugin:obsidian-git:21553
fulfilled @ plugin:obsidian-git:20402
```

```
[submodule "Projects/Personal/Steam Redeem/Steam Redeem Gist"]
url = https://gist.github.com/REDACTED.git
```

**Comment by AmauryVanEspen** [comment]
I've got the same issue.

**Reply by Vinzent03** [reply]
Please show your `.gitmodules` file and see [this](https://github.com/denolehov/obsidian-git/discussions/438#discussioncomment-6722558) new comment.

**Reply by AmauryVanEspen** [reply]
Hello Vincent03
I've read your new comment, I've the path and the URL.
[submodule "Sync-Marjorie-Amaury"]
	path = Sync-Marjorie-Amaury
	url = https://Personal_Token_Key@github.com/MainObsidianVault/Sync-Marjorie-Amaury.git
Initially I've used SubRepo before to use gitmodule
Cheers
Amaury

**Reply by Vinzent03** [reply]
So it still doesn't work? Please show your error message.

**Comment by Vinzent03** [comment]
I guess you added the submodule with a different path, because your path contains spaces, but the remote repo doesn't. When I add a submodule, the `.gitmodules` entry for that submodule contains a path value, but yours doesn't.

```
git submodule add git@github.com:Vinzent03/obsidian-testing.git "obsidian testing"
```
creates the following `.gitmodules` file:
```
[submodule "obsidian testing"]
        path = obsidian testing
        url = git@github.com:Vinzent03/obsidian-testing.git
```
